app/controllers/admin/users_controller.rb

```diff
@@ -1,15 +1,20 @@
 class Admin::UsersController < ApplicationController
+  before_action :find_user, only: %i[edit update destroy]
+
```

Comment:
how about putting authorize user to callback here and in other similar places

```
before_action -> { authorize :user }, only: %i[edit update destroy]
```
not tested but sth like above

Reply:
It would have needed to be before_action -> { authorize @user } in case the policy needed access to the resource itself, but have renaming find methods instead.